src/pdfxml.cpp

```diff
@@ -508,58 +508,363 @@ wxPdfTable::InsertCell(wxPdfTableCell* cell)
   }
 }
 
+double
+wxPdfTable::WriteOnPage(bool writeHeader, double x, double y)
+{
+  y = WriteRowsOnPage(m_bodyRowFirst, m_bodyRowLast, x, y, writeHeader);
+  return y;
+}
+
+double
+wxPdfTable::WriteRowsOnPage(unsigned firstRow, unsigned lastRow, double x, double y, bool writeHeader)
+{
+  if (writeHeader)
+  {
+    y = WriteRows(m_headRowFirst, m_headRowLast, x, y, true);
+  }
+  y = WriteRows(firstRow, lastRow, x, y, false);
+  return y;
+}
+
+double
+wxPdfTable::WriteOnMultiplePages(bool writeHeader, const wxArrayInt& firstRowsOnNextPage, double x, double y)
+{
+  wxArrayInt::const_iterator endIter = firstRowsOnNextPage.end();
+  unsigned int firstRow = m_bodyRowFirst;
+  unsigned int firstPageBreak = *firstRowsOnNextPage.begin();
+
+  if(firstPageBreak > firstRow)
+  {
+    y = WriteRowsOnPage(firstRow, firstPageBreak, x, y, writeHeader);
+  }
+
+  for(wxArrayInt::const_iterator iter = firstRowsOnNextPage.begin(); iter != endIter; ++iter)
+  {
+    m_document->AddPage(m_document->GetPageOrientation(), false);
+    y = m_document->GetY();
+    firstRow = *iter;
+    //Determine last table row on page
+    unsigned int lastRow = m_bodyRowLast;
+    wxArrayInt::const_iterator nextIter = iter + 1;
+    if(nextIter != endIter)
+    {
+      lastRow = (*nextIter);
+    }
+    //Write rows on page
+    y = WriteRowsOnPage(firstRow, lastRow, x, y, writeHeader);
+  }
+  return y;
+}
+
 void
 wxPdfTable::Write()
 {
-  bool writeHeader = m_headRowLast > m_headRowFirst;
-  bool newPage = false;
-  double saveLeftMargin = m_document->GetLeftMargin();
-  double x, y;
-  unsigned int row, headRow;
-  y = m_document->GetY();
-  double breakMargin = m_document->GetBreakMargin();
-  double pageHeight = m_document->GetPageHeight();
-  double yMax = pageHeight - breakMargin;
-  if (y + m_headHeight + m_maxHeights[m_bodyRowFirst] > yMax)
-  {
-    newPage = true;
+  const bool writeHeader = m_headRowLast > m_headRowFirst;
+  const double x = m_document->GetLeftMargin();
+  double y = m_document->GetY();
+
+  // check if table is about more pages
+  // if yes, the first body rows are in the returning array
+  const wxArrayInt firstRowsOnNextPage = GetFirstRowsOnNextPage();
+
+  //Case we get everything on the same Page
+  if(firstRowsOnNextPage.empty())
```

Comment:
Maybe it is not necessary to separately handle the case where the whole table fits on one page, if you add a row number 1 larger than `m_bodyRowLast` as a sentinel to `firstRowsOnNextPage`. This would require only a small change to method `wxPdfTable::WriteOnMultiplePages`, namely checking whether the first entry is not the sentinel, before issuing a page break.

Reply:
yes, there are different ways to solve it. I thought it could be more readable in that way than to enter a sentinel. How ever, somewhere it has to be checked if it's all on one page, even if it's done in ```GetFirstRowsOnNextPage```.

Another possibility would be to call ```WriteRowsOnPage(unsigned firstRow, unsigned lastRow, double x, double y, bool writeHeader)``` directly in the cited ```if```, which would make ```WriteOnPage``` obsolete.

Reply:
> yes, there are different ways to solve it. I thought it could be more readable in that way than to enter a sentinel. How ever, somewhere it has to be checked if it's all on one page, even if it's done in `GetFirstRowsOnNextPage`.

Of course, it has to be checked somewhere one way or the other, but using a sentinel entry in `firstRowsOnNextPage` makes this easy and eliminates any special handling.

> Another possibility would be to call `WriteRowsOnPage(unsigned firstRow, unsigned lastRow, double x, double y, bool writeHeader)` directly in the cited `if`, which would make `WriteOnPage` obsolete.

Well, actually I would prefer the solution with the sentinel entry. The only check that has to be made in `WriteOnMultiplePages` is when the sentinel is reached, so that not too many page breaks are issued. A method name like `WriteTable`, `WriteAll`, or `WriteAllRows` might be more appropriate than `WriteOnMultiplePages`, when using this approach.